apps/css-workshop/src/components/Button.astro

```diff
@@ -1,22 +1,40 @@
 ---
 type Props = {
-  variant?: 'high-visibility' | 'borderless' | 'cta';
+  variant?: 'high-visibility' | 'borderless' | 'cta' | 'idea';
   size?: 'small' | 'large';
-  active?: boolean;
-  shift?: 'left' | 'right';
-  iuiType?: 'feedback' | 'dropdown';
+  isActive?: boolean;
+  dropdown?: boolean;
 } & astroHTML.JSX.ButtonHTMLAttributes;
 
-const { class: className, variant, size, active, shift, iuiType, ...props } = Astro.props;
+const { class: className, variant, size, isActive, dropdown, ...props } = Astro.props;
 ---
 
 <button
-  class:list={['iui-button', { 'iui-button-dropdown': iuiType === 'dropdown' }, className]}
-  data-iui-variant={iuiType === 'feedback' ? 'idea' : variant}
+  class:list={['iui-button', 'iui-button-base', { 'iui-button-dropdown': dropdown }, className]}
+  type='button'
+  data-iui-variant={variant}
   data-iui-size={size}
-  data-iui-active={active ? 'true' : undefined}
-  data-iui-shift={shift}
+  data-iui-active={isActive ? 'true' : undefined}
   {...props}
 >
-  <slot />
+  <slot name='start-icon' />
+  {
+    Astro.slots.has('default') && (
+      <span>
+        <slot />
+      </span>
+    )
+  }
+  <slot name='end-icon' />
 </button>
+
+<script>
+  const buttonIcons = [
+    ...document.querySelectorAll<HTMLInputElement>('[slot="start-icon"]'),
+    ...document.querySelectorAll<HTMLInputElement>('[slot="end-icon"]'),
+    ...document.querySelectorAll<HTMLInputElement>('.iui-notification-marker'),
+  ];
+  buttonIcons.forEach((icon) => {
+    icon.classList.add('iui-button-icon');
+  });
+</script>
```

Comment:
You should never need a client-script to generate HTML. Instead, just create a wrapping span with `iui-button-icon` and `aria-hidden`, similar to how it is done in react (see [`Button.tsx`](https://github.com/iTwin/iTwinUI/blob/main/packages/itwinui-react/src/core/Buttons/Button.tsx#L89-L92)).

Reply:
Alright, will replace.

Reply:
Done in this [commit.](https://github.com/iTwin/iTwinUI/pull/2012/commits/b692e8a2547028a6c1c0ea1f40ade2c9795ffb0c)

Reply:
Why do you have `iui-button-icon` in some places on the button page?

Reply:
Removing the `iui-button-icon` class from the top level `iui-notification-marker` components caused the icon to shift up. However, testing again, it seems that the issue was being caused by removing that class while leaving the `iui-button-icon` on the interior `svg` since removing both does not cause the shift. Will update to remove the classes.

Reply:
Removed in this [commit.](https://github.com/iTwin/iTwinUI/pull/2012/commits/5fb02d4fabeede1ab5c06df2bb9c5b3fb5e156ec)